core/src/data/value/mod.rs

```diff
@@ -902,4 +962,27 @@ mod tests {
             .into()),
         );
     }
+
+    #[test]
+    fn test_unary_minus() {
+        assert_eq!(I8(1).unary_minus(), Ok(I8(-1)));
+        assert_eq!(I64(1).unary_minus(), Ok(I64(-1)));
+
+        assert_eq!(F64(1.0).unary_minus(), Ok(F64(-1.0)));
+        assert_eq!(
+            Decimal(Decimal::ONE).unary_minus(),
+            Ok(Decimal(-Decimal::ONE))
+        );
+
+        assert_eq!(
+            Str("abc".to_string()).unary_minus(),
+            Err(ValueError::UnaryMinusOnNonNumeric.into())
+        );
+    }
+
+    #[test]
+    fn test_factorial() {
```

Comment:
> unit tests already include the meaning of test. so, how about to use simply.. unary_minus() for this test name?

This, too.
It would be enough with
```rust
#[test]
fn factorial() {
 . .
```